xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -85,47 +85,81 @@ PerimeterGenerator::process()
             for (int i = 0; i <= loop_number+1; ++i) {  // outer loop is 0
                 Polygons offsets;
                 if (i == 0) {
-                    // the minimum thickness of a single loop is:
-                    // ext_width/2 + ext_spacing/2 + spacing/2 + width/2
+                    // compute next onion, without taking care of thin_walls : destroy too thin areas.
+                    if (!this->config->thin_walls)
+                        offsets = offset(last, -(float)(ext_pwidth / 2));
```

Comment:
is explicit cast needed?

Reply:
I wrote (float) to avoid an implicit conversion (i find them dangerous sometimes).